hazelcast/src/main/java/com/hazelcast/nio/Packet.java

```diff
@@ -88,12 +88,35 @@ public void setConn(Connection conn) {
         this.conn = conn;
     }
 
-    public void setFlag(int bit) {
-        flags |= 1 << bit;
+    /**
+     * Sets a particular flag.
```

Comment:
could you refine the contract a bit? something like: "other flags are left intact"